command/harness/dlite/dlite.go

```diff
@@ -103,29 +104,31 @@ func (c *dliteCommand) run(*kingpin.ParseContext) error {
 		cancel()
 	})
 
-	instanceStore, stageOwnerStore, err := database.ProvideStore(c.env.Database.Driver, c.env.Database.Datasource)
-	if err != nil {
-		logrus.WithError(err).Fatalln("Unable to start the database")
-	}
-
-	c.stageOwnerStore = stageOwnerStore
-	c.poolManager = drivers.New(ctx, instanceStore, &c.env)
+	// Initialize metrics
+	c.registerMetrics()
 
-	poolConfig, err := harness.SetupPool(ctx, &c.env, c.poolManager, c.poolFile)
-	defer harness.Cleanup(&c.env, c.poolManager) //nolint: errcheck
+	poolConfig, err := c.setupPool(ctx)
+	defer harness.Cleanup(&c.env, c.poolManager, true, true) //nolint: errcheck
 	if err != nil {
-		logrus.WithError(err).Error("could not setup pool")
 		return err
 	}
 
+	if env.Database.Postgres.Enabled {
```

Comment:
I feel like now is the right time to move to `wire` in this project and do dependency injection. Maybe we can do it subsequently but all this would be so much easier to manage through wire.